### PR TITLE
Fix account profiling incumbency

### PR DESCRIPTION
**Pull request by gritukan**
Current implementation is racy for two reasons.

Firstly, the following scenario is possible:
1) Peer is a leader and profiles metrics
2) Leader is switched, profiling invoker is destroyed
3) Dynamic config changes, StartAccountsProfiling is called and new (epoch) invoker in created
4) Peer becomes leader again, invoker is not recreated while it is still bound to the finished epoch

Secondly, invoker is always stopped when some incumbency finishes, while it is possible that only a part of incumbencies should be finished (i.e. during the change of number of alive followers).

This PR fixes these problems and also does some code cleanup.

### CODE REVIEW

File: yt/yt/server/master/security_server/security_manager.cpp [hunk 1]
```diff
@@ -4875,16 +4881,12 @@ class TSecurityManager
...
+        MaybeToggleAccountsProfiling();
 
-            if (newConfig->AccountsProfilingPeriod != oldConfig->AccountsProfilingPeriod &&
-                AccountsProfilingExecutor_)
-            {
-                AccountsProfilingExecutor_->SetPeriod(newConfig->AccountsProfilingPeriod);
-            }
-        } else {
-            StopAccountsProfiling();
+        if (newConfig->AccountsProfilingPeriod != oldConfig->AccountsProfilingPeriod &&
```
[the remark this answers is not in the document]
Comment (gritukan) [reply yt/yt/server/master/security_server/security_manager.cpp:4886]: Seems like it's redundant, will remove

**Review by raundn — APPROVED** [review]

**Comment by raundn** [comment]
Shipped!

**Review by Kontakter — APPROVED** [review]

**Comment by Kontakter** [comment]
There is a compilation error:
```
yt/yt/server/master/security_server/security_manager.cpp:4869:21: error: unused variable 'oldConfig' [-Werror,-Wunused-variable]
        const auto& oldConfig = oldClusterConfig->SecurityManager;
                    ^
1 error generated.
```

**Comment by gritukan** [comment]
Oops, sorry, I don't know how it compiled on my side. Fixed.

**Comment by raundn** [comment]
Shipped!

**Comment by raundn** [comment]
Shipped!

**Comment by raundn** [comment]
Shipped!

**Comment by raundn** [comment]
Shipped!

**Comment by raundn** [comment]
shipped!

**Pull request closed without merging**